modules/frontend/hedged_requests.go

```diff
@@ -0,0 +1,53 @@
+package frontend
+
+import (
+	"net/http"
+	"time"
+
+	"github.com/cristalhq/hedgedhttp"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promauto"
+)
+
+const (
+	hedgedMetricsPublishDuration = 10 * time.Second
+)
+
+var (
+	hedgedRequestsMetrics = promauto.NewGauge(
+		prometheus.GaugeOpts{
+			Namespace: "tempo",
+			Name:      "query_frontend_hedged_roundtrips_total",
+			Help:      "Total number of hedged search requests. Registered as a gauge for code sanity. This is a counter.",
+		},
+	)
+)
+
+func newHedgedRequestWare(cfg HedgingConfig) Middleware {
+	return MiddlewareFunc(func(next http.RoundTripper) http.RoundTripper {
+		if cfg.HedgeRequestsAt == 0 {
+			return next
+		}
+		ret, stats, err := hedgedhttp.NewRoundTripperAndStats(cfg.HedgeRequestsAt, cfg.HedgeRequestsUpTo, next)
```

Comment:
should we construct this once on line 27  and then use it repeatedly?

Reply:
I had the same doubt while coding it, but this is only called once when instantiating the trace by ID round tripper if I'm not mistaken.

Reply:
you are correct on both this and the below comment. i was confused by the multiple function nesting :+1: